src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs

```diff
@@ -96,106 +96,123 @@ public void Build(IList<DocumentBuildParameters> parameters, string outputDirect
-                {
-                    // Save to manifest.json
                     EnvironmentContext.FileAbstractLayerImpl =
                         FileAbstractLayerBuilder.Default
-                        .ReadFromRealFileSystem(parameters[0].OutputBaseDir)
-                        .WriteToRealFileSystem(parameters[0].OutputBaseDir)
+                        .ReadFromManifest(generatedManifest, parameters[0].OutputBaseDir)
+                        .WriteToManifest(generatedManifest, parameters[0].OutputBaseDir)
                         .Create();
-                    SaveManifest(generatedManifest);
-                }
+                    using (new PerformanceScope("Process"))
+                    {
+                        _postProcessorsManager.Process(generatedManifest, outputDirectory);
+                    }
 
-                using (new PerformanceScope("Cleanup"))
-                {
-                    EnvironmentContext.FileAbstractLayerImpl = null;
+                    using (new PerformanceScope("Dereference"))
+                    {
+                        generatedManifest.Dereference(parameters[0].OutputBaseDir, parameters[0].MaxParallelism);
+                    }
 
-                    // overwrite intermediate cache files
-                    if (_intermediateFolder != null && transformDocument)
+                    using (new PerformanceScope("SaveManifest"))
                     {
-                        _currentBuildInfo.Save(_intermediateFolder);
-                        if (_lastBuildInfo != null)
+                        // Save to manifest.json
+                        EnvironmentContext.FileAbstractLayerImpl =
+                            FileAbstractLayerBuilder.Default
+                            .ReadFromRealFileSystem(parameters[0].OutputBaseDir)
+                            .WriteToRealFileSystem(parameters[0].OutputBaseDir)
+                            .Create();
+                        SaveManifest(generatedManifest);
+                    }
+
+                    using (new PerformanceScope("Cleanup"))
+                    {
+                        EnvironmentContext.FileAbstractLayerImpl = null;
+
+                        // overwrite intermediate cache files
+                        if (_intermediateFolder != null && transformDocument)
                         {
-                            Directory.Delete(Path.Combine(Environment.ExpandEnvironmentVariables(_intermediateFolder), _lastBuildInfo.DirectoryName), true);
+                            _currentBuildInfo.Save(_intermediateFolder);
+                            if (_lastBuildInfo != null)
+                            {
+                                try
+                                {
+                                    Directory.Delete(Path.Combine(Environment.ExpandEnvironmentVariables(_intermediateFolder), _lastBuildInfo.DirectoryName), true);
+                                }
+                                catch (Exception)
```

Comment:
IOException

---
In reply to: [106574086](https://github.com/dotnet/docfx/pull/1429#discussion_r106574086) [](ancestors = 106574086)